MathLib/LinAlg/RowColumnIndices.h

```diff
@@ -14,19 +14,25 @@
 
 namespace MathLib
 {
+// A dummy vector as the default value of the third argument of
+// the constructor.
+static std::vector<std::size_t> dummy_non_ghost_local_ids;
 
 template <typename IDX_TYPE>
 struct RowColumnIndices
 {
 	typedef typename std::vector<IDX_TYPE> LineIndex;
-	RowColumnIndices(LineIndex const& rows_, LineIndex const& columns_)
-		: rows(rows_), columns(columns_)
+	RowColumnIndices(LineIndex const& rows_, LineIndex const& columns_,
```

Comment:
At least you could make `dummy_ghost_local_ids` static const, or you make a pointer out of `ghost_local_ids`. Sure, that would be somehow inconsistent as the other members are references, but it removes the need for a static dummy constant.